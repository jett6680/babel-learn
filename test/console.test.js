const path = require('path')
const pluginTester = require('babel-plugin-tester').default
const printLogInfoPlugin = require('../plugin/babel-plugin-print-log')

pluginTester({
  filename: 'console.test.js',
  plugin: printLogInfoPlugin,
  fixtures: path.join(__dirname, '__fixtures__'),
  tests: {
    'add console log': {
      code: 'console.log(1111)',
      output: `console.log(\"unkown filename: (1, 0)\");\nconsole.log(1111);`
    },
    'test snapshot': {
      code: `
        function sayHi(person) {
          console.log('nihao')
          return 'Hello ' + person + '!'
        }
      `,
      snapshot: true,
    },
    'test input file': {
      fixture: path.join(__dirname, './__fixtures__/changed.js'),
      outputFixture: path.join(__dirname, './__fixtures__/changed-output.js')
    }
  }
})