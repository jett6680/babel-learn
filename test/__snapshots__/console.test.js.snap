// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin test snapshot: test snapshot 1`] = `

function sayHi(person) {
  console.log('nihao')
  return 'Hello ' + person + '!'
}

      ↓ ↓ ↓ ↓ ↓ ↓

function sayHi(person) {
  console.log("unkown filename: (2, 2)");
  console.log("nihao");
  return "Hello " + person + "!";
}


`;
